vpr/src/noc/read_xml_noc_traffic_flows_file.cpp

```diff
@@ -101,18 +101,69 @@ void process_single_flow(pugi::xml_node single_flow_tag, const pugiutil::loc_dat
     check_traffic_flow_router_module_type(sink_router_module_name, sink_router_id, single_flow_tag, loc_data, cluster_ctx, noc_router_tile_type);
 
     // store the properties of the traffic flow
-    double traffic_flow_bandwidth = pugiutil::get_attribute(single_flow_tag, "bandwidth", loc_data, pugiutil::REQUIRED).as_double(NUMERICAL_ATTRIBUTE_CONVERSION_FAILURE);
+    double traffic_flow_bandwidth = get_traffic_flow_bandwidth(single_flow_tag, loc_data);
 
-    double max_traffic_flow_latency = pugiutil::get_attribute(single_flow_tag, "latency_cons", loc_data, pugiutil::REQUIRED).as_double(NUMERICAL_ATTRIBUTE_CONVERSION_FAILURE);
+    double max_traffic_flow_latency = get_max_traffic_flow_latency(single_flow_tag, loc_data);
 
-    verify_traffic_flow_properties(traffic_flow_bandwidth, max_traffic_flow_latency, single_flow_tag, loc_data);
+    int traffic_flow_priority = get_traffic_flow_priority(single_flow_tag, loc_data);
+
+    verify_traffic_flow_properties(traffic_flow_bandwidth, max_traffic_flow_latency, traffic_flow_priority, single_flow_tag, loc_data);
 
     // The current flow information is legal, so store it
-    noc_traffic_flow_storage->create_noc_traffic_flow(source_router_module_name, sink_router_module_name, source_router_id, sink_router_id, traffic_flow_bandwidth, max_traffic_flow_latency);
+    noc_traffic_flow_storage->create_noc_traffic_flow(source_router_module_name, sink_router_module_name, source_router_id, sink_router_id, traffic_flow_bandwidth, max_traffic_flow_latency, traffic_flow_priority);
 
     return;
 }
 
+double get_traffic_flow_bandwidth(pugi::xml_node single_flow_tag, const pugiutil::loc_data& loc_data) {
+    double traffic_flow_bandwidth;
+    // holds the bandwidth value as a string so that it can be used to convert to a floating point value (this is done so that scientific notation is supported)
+    std::string traffic_flow_bandwidth_intermediate_val = pugiutil::get_attribute(single_flow_tag, "bandwidth", loc_data, pugiutil::REQUIRED).as_string();
+
+    // now convert the value to double
+    traffic_flow_bandwidth = std::atof(traffic_flow_bandwidth_intermediate_val.c_str());
+
+    return traffic_flow_bandwidth;
+}
+
+double get_max_traffic_flow_latency(pugi::xml_node single_flow_tag, const pugiutil::loc_data& loc_data) {
+    // default latency constraint is the maximum double val (indicating that there is no constraint)
```

Comment:
Update comment.  May just want to say "set to large value, indicating no constraint".

Reply:
Updated